### ISSUE TITLE
Scenario analysis spits error for no obvious reason

### ISSUE THREAD

**Issue by DenoBeno** [clarity-h2020/scenario-analysis#17]
This is what I see on https://csis.myclimateservice.eu/study/35/step/1528/view/scenario-analysis

![image](https://user-images.githubusercontent.com/2961790/66201791-667cf100-e6a4-11e9-92e3-b5b9776db35b.png)

Error message seems to be bogus.

**Comment by DenoBeno** [comment]
Hm wait, the data in question is really not shown in the table. However, the bar charts are missing some other data too, e.g. rcp 26 for 2071-2100 period.

the table component is still not showing anything, so I can't say what is really wrong here. https://csis.myclimateservice.eu/study/35/step/1528/view/table

**Comment by p-a-s-c-a-l** [comment]
Yes, this error occurs *from time to time* because the data retrieved from the [EMIKAT RA/IA API](https://github.com/clarity-h2020/emikat/issues/9) does not contain all values. I don't know why this happens. BTW, this is different view (https://service.emikat.at/EmiKatTst/api/scenarios/3189/feature/view.2994/table/data) than the one used by the table component. It **aggregates** the values used by the table component. So maybe there's simply no result for 

- RCP26 (2041-2070), Event Frequency: Rare
- RCP45 (2011-2040). Event Frequency: Frequent

available, for scenario 3189? Maybe @humerh can have a look?

**Comment by humerh** [comment]
Yes, you are right. These Events does not exist.

See the table:

1201 | 37,98 | 23,73 |   |   | 1 | HW | Baseline | Baseline | Occasional | 38_0.366d | 34,7
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1192 | 37,98 | 23,73 |   |   | 1 | HW | Baseline | Baseline | Rare | 39.5_0.066d | 34,7
1210 | 37,98 | 23,73 |   |   | 1 | HW | Baseline | Baseline | Frequent | 35.5_3.133d | 34,7
1193 | 37,98 | 23,73 |   |   | 1 | HW | rcp26 | 20110101-20401231 | Rare | 40_0.133d | 34,7
1211 | 37,98 | 23,73 |   |   | 1 | HW | rcp26 | 20110101-20401231 | Frequent | 36.5_3.233d | 34,7
1202 | 37,98 | 23,73 |   |   | 1 | HW | rcp26 | 20110101-20401231 | Occasional | 39.5_0.266d | 34,7
1212 | 37,98 | 23,73 |   |   | 1 | HW | rcp26 | 20410101-20701231 | Frequent | 37.5_2.066d | 34,7
1203 | 37,98 | 23,73 |   |   | 1 | HW | rcp26 | 20410101-20701231 | Occasional | 40.5_0.366d | 34,7
1205 | 37,98 | 23,73 |   |   | 1 | HW | rcp26 | 20710101-21001231 | Occasional | 41_0.300d | 34,7
1214 | 37,98 | 23,73 |   |   | 1 | HW | rcp26 | 20710101-21001231 | Frequent | 37_3.333d | 34,7
1195 | 37,98 | 23,73 |   |   | 1 | HW | rcp26 | 20710101-21001231 | Rare | 42_0.133d | 34,7
1199 | 37,98 | 23,73 |   |   | 1 | HW | rcp45 | 20110101-20401231 | Occasional | 40.5_0.266d | 34,7
1191 | 37,98 | 23,73 |   |   | 1 | HW | rcp45 | 20410101-20701231 | Rare | 43.5_0.066d | 34,7
1200 | 37,98 | 23,73 |   |   | 1 | HW | rcp45 | 20410101-20701231 | Occasional | 41_0.266d | 34,7
1209 | 37,98 | 23,73 |   |   | 1 | HW | rcp45 | 20410101-20701231 | Frequent | 38_2.133d | 34,7
1206 | 37,98 | 23,73 |   |   | 1 | HW | rcp45 | 20710101-21001231 | Occasional | 42_0.266d | 34,7
1196 | 37,98 | 23,73 |   |   | 1 | HW | rcp45 | 20710101-21001231 | Rare | 42.5_0.066d | 34,7
1215 | 37,98 | 23,73 |   |   | 1 | HW | rcp45 | 20710101-21001231 | Frequent | 38_3.400d | 34,7
1194 | 37,98 | 23,73 |   |   | 1 | HW | rcp85 | 20110101-20401231 | Rare | 39.5_0.133d | 34,7
1204 | 37,98 | 23,73 |   |   | 1 | HW | rcp85 | 20110101-20401231 | Occasional | 39_0.333d | 34,7
1213 | 37,98 | 23,73 |   |   | 1 | HW | rcp85 | 20110101-20401231 | Frequent | 37_2.300d | 34,7
1216 | 37,98 | 23,73 |   |   | 1 | HW | rcp85 | 20410101-20701231 | Frequent | 39_2.466d | 34,7
1197 | 37,98 | 23,73 |   |   | 1 | HW | rcp85 | 20410101-20701231 | Rare | 42.5_0.133d | 34,7
1207 | 37,98 | 23,73 |   |   | 1 | HW | rcp85 | 20410101-20701231 | Occasional | 41.5_0.600d | 34,7
1217 | 37,98 | 23,73 |   |   | 1 | HW | rcp85 | 20710101-21001231 | Frequent | 41_2.733d | 34,7
1198 | 37,98 | 23,73 |   |   | 1 | HW | rcp85 | 20710101-21001231 | Rare | 45.5_0.066d | 34,7
1208 | 37,98 | 23,73 |   |   | 1 | HW | rcp85 | 20710101-21001231 | Occasional | 44_0.266d | 34,7

**Comment by p-a-s-c-a-l** [comment]
> These Events does not exist.

Any why? Is that a problem with the input hazard data or something else?

**Comment by humerh** [comment]
This question has to be answered by Robert (@RobAndGo). 
I suppose, that there are not enough statistical data inputs available for this event.

**Comment by DenoBeno** [comment]
A, so we may very easily miss the 2-year events, I see. But I don't think that this is the issue we have at hand. Look at https://github.com/clarity-h2020/scenario-analysis/issues/17#issue-502564514 again. The whole RPC26 is missing for the period 2041-2070, not just the rare event.

Unless the lack of the "rare" event makes the generation fail for the others too.

**Comment by DenoBeno** [comment]
By the way: it would be good if the scenario analysis app would only show the data for the "presets" (we can allow users to define more than one preset). In that way, there would be less data sets to show and the labels would be easier for the users since users can freely choose them.

![image](https://user-images.githubusercontent.com/2961790/66472501-aa4d6d00-ea8d-11e9-8f0a-5df3dd86cfa9.png)

**Comment by RobAndGo** [comment]
OK, now I think I understand - you mean that there is no data from the RCP26 2041-2070 scenario. I thought I uploaded all the files in a previous github post - I'll try to find it now to see if maybe I forgot this one.

I can't seem to find the original post, so I'll upload the missing dataset here:
[europe_rcp26_20410101-20701231.csv.txt](https://github.com/clarity-h2020/scenario-analysis/files/3716853/europe_rcp26_20410101-20701231.csv.txt)
It's a csv file so one just needs to remove the .txt extension.

As an aside, does anyone know how to search on github for particular posts, e.g. I wanted to search for posts containing the links for these data files which I previously posted.

**Comment by p-a-s-c-a-l** [comment]
> It's a csv file so one just needs to remove the .txt extension.

OK. And who is responsible for making the data available via a service? Is it METEOGRID or @humerh? Is it directly imported into EMIKAT or fist into geo server then queried by EMIKAT?


> As an aside, does anyone know how to search on github for particular posts, e.g. I wanted to search for posts containing the links for these data files which I previously posted.

@RobAndGo [here](https://github.com/search?q=org%3Aclarity-h2020++europe_rcp26_20410101-20701231.csv.txt&type=Issues)

**Comment by DenoBeno** [comment]
Is this done now?

**Comment by p-a-s-c-a-l** [comment]
> Is this done now?

No: "Could not load indicator file RCP45 (2011-2040). It contains no indicator data for indicatorFrequent"

**Comment by RobAndGo** [comment]
@p-a-s-c-a-l do you mean this is a problem with the original data set that I uploaded, because I do have frequent heat wave events for RCP45 (2011-2040), here is the file once again:
[europe_rcp45_20110101-20401231.csv.txt](https://github.com/clarity-h2020/scenario-analysis/files/3742753/europe_rcp45_20110101-20401231.csv.txt)

**Comment by p-a-s-c-a-l** [comment]
I don't know. This data comes from the [EMIKAT RA/IA API](https://github.com/clarity-h2020/emikat/issues/9#issuecomment-536533426), so maybe @humerh knows why the events are still missing.

**Comment by DenoBeno** [comment]
@humerh , WDYT?

**Comment by p-a-s-c-a-l** [comment]
Events are still [missing](https://csis.myclimateservice.eu/study/35/step/1528/view/scenario-analysis).
Bug or Feature?

**Comment by p-a-s-c-a-l** [comment]
still not resolved yet: `Could not load indicator file RCP45 (2011-2040). It contains no indicator data for indicatorFrequent`

**Comment by p-a-s-c-a-l** [comment]
Could not load indicator file RCP45 (2011-2040). It contains no indicator data for indicatorFrequent
